Rules/UseLiteralInitializerForHashtable.cs

```diff
@@ -0,0 +1,401 @@
+            }
+
+            // no arguments provided to new OR one of the argument ends with ignorecase
+            // (heuristics find to something like [system.stringcomparer]::ordinalignorecase)
+            if (methodCallAst.Arguments == null
+                || !HasIgnoreCaseComparerArg(methodCallAst.Arguments))
+            {
+                var dr = new DiagnosticRecord(
+                    Strings.UseLiteralInitilializerForHashtableDescription,
+                    methodCallAst.Extent,
+                    GetName(),
+                    GetDiagnosticSeverity(),
+                    fileName,
+                    ruleId: null,
+                    suggestedCorrections: GetSuggestedCorrections(methodCallAst, this.fileName));
+                diagnosticRecords.Add(dr);
+            }
+
+            return AstVisitAction.Continue;
+        }
+
+        /// <summary>
+        /// Analyzes command ast to check for new-object command and parse its arguments
+        /// </summary>
+        private void AnalyzeNewObjectCommand(CommandAst commandAst)
+        {
+            string typeName;
+            List<string> argumentList;
+            GetParametersFromCommandAst(commandAst, out typeName, out argumentList);
+            if (typeName == null
+                || !presetTypeNameSet.Contains(typeName))
+            {
+                return;
+            }
+
+            if (argumentList != null)
+            {
+                if (argumentList.Any(arg => arg != null && arg.EndsWith("ignorecase", StringComparison.OrdinalIgnoreCase)))
+                {
+                    return;
+                }
+            }
+
+            var dr = new DiagnosticRecord(
+                Strings.UseLiteralInitilializerForHashtableDescription,
+                commandAst.Extent,
+                GetName(),
+                GetDiagnosticSeverity(),
+                fileName,
+                ruleId: null,
+                suggestedCorrections: GetSuggestedCorrections(commandAst, this.fileName));
+            diagnosticRecords.Add(dr);
+        }
+
+        /// <summary>
+        /// Interpret the named and unnamed arguments and assign them their corresponding parameters
+        /// </summary>
+        /// <param name="commandAst">An non-null instance of CommandAst. Expects it be commandast of "new-object" command</param>
+        /// <param name="typeName">Returns the TypeName argument</param>
+        /// <param name="argumentList">Returns the ArgumentList argument</param>
+        /// This should read the command in all the following form
+        /// new-object hashtable
+        /// new-object -Typename hashtable
+        /// new-object hashtable -ArgumentList comparer
+        /// new-object -Typename hashtable -ArgumentList blah1,blah2
+        /// new-object -ArgumentList blah1,blah2 -typename hashtable
+        private void GetParametersFromCommandAst(CommandAst commandAst, out string typeName, out List<string> argumentList)
```

Comment:
In V4, we added an API to help this scenario, see `System.Management.Automation.Language.StaticParameterBinder.BindCommand`.